### DISCUSSION TITLE
Rules clarification (always rejecting the insert)

### DISCUSSION THREAD

**Discussion by a4amado** [pocketbase/pocketbase#6115] [Q&A] [answered]
both `@request.auth.id ?= users ` and `users  ?= @request.auth.id` always reject

according to the docs `?=` `Any/At least one of Equal`

`users` will be a list of strings containing id's of the users that are part of this group,
Im trying to prevent user from creating a group unless he is part of this group

```tsx

export default function Home() {
  return (
    <div>
      <button
        onClick={() => {
          // @request.auth.id != "" && @request.body.users:length = 2 && (@request.body.users:each ?= @request.auth.id)
          pb.collection(Collections.Users)
            .authWithPassword("test@test.com", "test@test.com")
            .then(() => {
              pb.collection(Collections.Chat).create({
                users: ["83e99254660gv22", "83e99254660gv22"],
              } as ChatRecord);
            });
        }}
      >
        ss
      </button>
    </div>
  );
}

```

**Comment by ganigeorgiev** [comment, marked as answer]
Since `users` is a multiple `relation` field, then in this case `@request.auth.id ?= users` doesn't work because you are comparing the authenticated user id against the serialized version of the users value (multi-valued fields are stored in the database as serialized JSON array).

If you want the rule to apply against a field (in your case id) of the users then you have to use dot-notation to target the specific field, aka. `@request.auth.id ?= users.id` (or `:each` as in your body example).

NB! This rule doesn't prevent the user to remove or add other users from the "chat" collection because they can change the relation. If that's not the intended behavior, then you'll have to change your collections and use a more traditional SQL structure, e.g. store the relation between the 2 colletions in a separate junction collection that will have 2 single `relation` fields pointing back to the related user and chat.

If that doesn't help, please try to elaborate a little more on your collections structure (_you can also export it as JSON from the Dashboard > Settings > Export collections_) and the intended behavior.

**Comment by a4amado** [comment]
Oh Thanks, adding `:each` worked for me.
